azure/datalake/store/transfer.py

```diff
@@ -382,6 +393,17 @@ def _rename_file(self, src, dst, overwrite=False):
     
         logger.debug('Renamed %r to %r', src, dst)
 
+    def _update_progress(self, length):
+        if self._progress_callback is not None:
+            if self._progress_lock is not None:
+                with self._progress_lock:
+                    self._progress_total_bytes += length
+                    total = self._progress_total_bytes
```

Comment:
Why the local assignment and not just pass `self._progress_total_bytes` to the callback? Also, [assignment is atomic in CPython](https://docs.python.org/2/faq/library.html#what-kinds-of-global-value-mutation-are-thread-safe).